client/history/client.go

```diff
@@ -854,16 +859,22 @@ func (c *clientImpl) GetReplicationMessages(
 	close(respChan)
 	close(errChan)
 
+	var err error
+	if len(errChan) > 0 {
+		err = <-errChan
+	}
+
 	response := &types.GetReplicationMessagesResponse{MessagesByShard: make(map[int32]*types.ReplicationMessages)}
+	responseSize := int64(0)
 	for resp := range respChan {
 		for shardID, tasks := range resp.MessagesByShard {
+			responseSize += tasks.GetSizeInByte()
+			if responseSize > int64(c.rpcMaxSizeInBytes()) {
```

Comment:
Is `responseSize` here for a batch of tasks? If so, if batch size is too high, we still need to manually adjust the number otherwise it will still get blocked?

Reply:
we can adjust the dynamic config rpcMaxSizeInBytes